pywisetransfer/__init__.py

```diff
@@ -10,13 +13,25 @@ def add_resources(self):
         self.subscriptions = Subscription(client=self)
         self.users = User(client=self)
 
-    def __init__(self, api_key, environment="sandbox"):
+    def __init__(
+        self, api_key, environment="sandbox", private_key_file=None, private_key=None
```

Comment:
Maybe this is a pain, but I wonder if we could get away with only supporting a single argument (`private_key_file`), and reading-and-signing on-demand.

The idea being to minimize the duration of time that the private key is in memory within the process.

If the code to perform the retrieval of the key could move into a `pywisetransfer.request_signing` module too, I think that'd make me happier.  The thinking there is to keep most (if not all) of the cryptography-related code (and imports) in one place with the relevant imports.  I don't know why I care so much about that but a) it mirrors the `webhooks` module and b) it keeps the imports in this `__init__.py` file minimal (and maybe c) it keeps the private-key handling code fairly minimal and hopefully easier to spot problems with - not that this file is huge or particularly distracting content-wise).

Reply:
I was tempted to only allow `private_key_file`, but I can guarantee people will want to get the key from elsewhere (e.g. vault or metadata). I also considered a callback, but I wanted the golden path to be as easy as possible.

I'm not sure minimising the amount of time the private key is in memory is worthwhile, as python doesn't have its own mechanism for marking memory as sensitive so it's likely to remain around anyway. It'll also be loaded back in every 10 minutes or so, when the 2FA validity expires.

Reply:
Yep, all seems reasonable.  I guess what I'm more worried about rather than "can Python mark memory as sensitive today" is "in five years will storing keys on the `Client` instance be challenging to unpick".  At the moment the distance-to-refactor to change that is small, so it's fine - but it's tricky to track that distance metric actively.

Fairly likely to merge this soon, I'll take another review pass and spend a bit of time thinking in the meantime.